### DISCUSSION TITLE
Accept GPIO for input?

### DISCUSSION THREAD

**Discussion by rjadams82** [thomst08/OctoLight#9] [Ideas]
I wonder if you are able to poll GPIO to toggle the octolight state?

May be helpful to explain my use case. I am controlling a light assembly with Octolight. I’m not using “light is a button” for the light output, just controlling a relay for steady on or off. When I walk up to my printer it’s helpful to have a physical switch to control the light if I don’t have my laptop. 

If I can wire a momentary push button to a different GPIO, can Octolight take that as input to toggle to the output state?

**Comment by thomst08** [comment]
Hey @rjadams82,

That's not a bad idea, honestly I have thought about it but not felt the need, my setup is a relay like yours, but I have a switch on the other side with the light, so I can switch it and this overrides the input from the relay.

As for doing it through the pi.. There's no reason why it couldn't be done... I want to make sure this adds value to the plugin rather then just filling it features, but I don't think this is too much of an issue and does bring some value in some situations.
I will have a think and look into how it could be done.

Good idea :)

**Reply by rjadams82** [reply]
The manual switch was a thought but knowing it would defeat the OctoLight input my mind started conceiving more complicated solutions as it always does. Not a priority 1 thing for sure just an idea.
Appreciate your prompt responses.

**Reply by thomst08** [reply]
That's a fair call, there is one thing that I question.
1. Would this switch be a constant on or a toggle on (like a button press)
2. If its a constant on, how would OctoLight to react if the light was already on? Or gets turned off in the web portal after being turned on with the switch?

**Reply by rjadams82** [reply]
I don’t think a two position switch with constant on/off will work well with the Octolight state toggle. 

I will use a momentary push button wired to a different GPIO and system ground. Pushing the button will momentarily bring the GPIO to ground. Will that work?

**Reply by thomst08** [reply]
I agree with you, that's the way I would wire it up.
But I want to make sure that it would work for other setups and is flexible there, it might be just worth treating it as a button press to change the sate on the PI then worrying too much about a switch setup.

I'll work this with and see what I can come up with.

**Reply by thomst08** [reply]
Hey @rjadams82, just an update here, I haven't forgotten about this, just been busy...
I have started toying with the code again and have setup a function to handle a external button to be linked up on another pin and connected to group or a v5 pin (depending on the settings you set).  When the button is pressed it will change the state of the light.

I am pretty happy with the setup, but I want to clean up the settings menu and a few other things before I release it.  Not far off, but just keeping you updated.

**Reply by thomst08** [reply]
Hey @rjadams82,

I have released a new version (v1.0.0), this allow for an external button to be added, enable the setting and set you pin, enable the v5 setting if you have the button connected to the v5 pin or leave it disabled if its connected to ground.
Please have a look and let me know if you have any issues.
